examples/cns-sfnettest.yaml

```diff
@@ -51,8 +51,10 @@ metadata:
 spec:
   restartPolicy: Always
   securityContext:
-    allowPrivilegeEscalation: false
-    runAsUser: 0
+    runAsUser: 1000
+    runAsNonRoot: true
+    seccompProfile:
+      type: RuntimeDefault
```

Comment:
I'm surprised some of this isn't the default? Same for later snippets.

Reply:
OC demands (some of) these security fields in the manifests. I ran a small experiment to verify if all of these were needed with `oc apply` with and without `--dry-run=server` and discovered a couple we can remove! So, thanks for raising the concern. 👍

- Without `spec.securityContext.runAsNonRoot: true`:
```
Warning: would violate PodSecurity "restricted:latest": runAsNonRoot != true (pod or container "onload-sfnettest-client" must set securityContext.runAsNonRoot=true)
```

- Without `spec.securityContext.seccompProfile.type: RuntimeDefault`:
```
Warning: would violate PodSecurity "restricted:latest": seccompProfile (pod or container "onload-sfnettest-client" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

- Without `spec.containers.[].securityContext.capabilities.drop: [ALL]`:
```
Warning: would violate PodSecurity "restricted:latest": unrestricted capabilities (container "onload-sfnettest-client" must set securityContext.capabilities.drop=["ALL"])
```

- Without `spec.containers.[].securityContext.allowPrivilegeEscalation: false`:
```
Warning: would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "onload-sfnettest-client" must set securityContext.allowPrivilegeEscalation=false)
```

- Without `spec.securityContext.runAsUser: 1000`: no complaints!
- Without `spec.containers.[].securityContext.privileged: false`: no complaints!

Additionally, it matches with the template pod suggested by OCP UI in the "Create pod" view:
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: example
  labels:
    app: httpd
  namespace: default
spec:
  securityContext:
    runAsNonRoot: true
    seccompProfile:
      type: RuntimeDefault
  containers:
    - name: httpd
      image: 'image-registry.openshift-image-registry.svc:5000/openshift/httpd:latest'
      ports:
        - containerPort: 8080
      securityContext:
        allowPrivilegeEscalation: false
        capabilities:
          drop:
            - ALL
```

I'm removing these unneeded fields from the manifests in the upcoming force push.

Reply:
The extra security overrides are required due to the presence of:
```yaml
metadata:
  namespace: default
```
Removing or otherwise changing the namespace to almost anything else (eg. `onload-runtime`) removes the security violation warning.

Reply:
That is a good find, thank you. (Also, for checking in a test cluster with `onload-sfnettest-client-pcolledg-bb9b4b4b6-fhbdd`.)

Let me move them into `onload-examples` and remove the remaining security fields in the upcoming git-force-push.

Reply:
Actually, let me keep that as-is. As discussed with @pcolledg-xilinx offline, the new "onload-examples" won't address the admission problem because new namespaces are created "restricted", and we would need to relax that constraint if we want to remove the security-related fields. Filed ON-15013 to drill into the security aspect more widely.

Reply:
We need to document OCP v4.10 exceptions (ie. remove `seccompProfile:`)

Reply:
OCP v4.10 seems to define `seccompProfile` along with `RuntimeDefault` in https://docs.openshift.com/container-platform/4.10/rest_api/workloads_apis/pod-v1.html.